### PR TITLE
Os locator Integration

### PR DESCRIPTION
**Pull request by piercetrey-figure**
- [x] register affiliate with new object store locator service upon creation
  - [x] determine which key to register (signing/encryption/both?)
- [x] switch from snapshot to initial published version of os-locator-sdk

### CODE REVIEW

File: versions.gradle [hunk 1]
```diff
@@ -20,6 +20,7 @@ ext {
     datadog_version = '2.11.0'
     logback_version = '1.2.3'
     javax_annotation_version = '1.3.2'
+    os_locator_sdk_version = '1.0-SNAPSHOT'
```
Comment (piercetrey-figure) [inline versions.gradle:23]: oh, @scirner22 this was the other thing I am waiting on, I have [this ticket in](https://app.clubhouse.io/figure/story/102372/upgrade-ossrh-secrets-from-p8e-repository-level-to-provenance-io-organization-level) to get the OSSRH secrets moved to the org level so I can publish the os-locator-sdk to Maven Central... I'll see if they will actually get to it today

**Comment by piercetrey-figure** [comment]
@rchaing-figure @scirner22 probably should get this merged in if we are in a good spot to do so, so I don't forget about it :badpokerface:... I think some of this stuff will have to be brought over to the new sdk as well, but need to think about how this flow will work (probably just have some call for the client to register an affiliate with object-store/the chain and make them manage any failures around that)

File: p8e-api/src/main/kotlin/io/provenance/engine/service/JobService.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import p8e.Jobs
+import java.lang.IllegalArgumentException
+import java.util.concurrent.CompletableFuture
+import kotlin.concurrent.thread
+
+@Component
+class JobService(private val jobHandlerServiceFactory: JobHandlerServiceFactory) {
```
Comment (rchaing-figure) [inline p8e-api/src/main/kotlin/io/provenance/engine/service/JobService.kt:19]: what is the job service actually doing?

Comment (piercetrey-figure) [reply p8e-api/src/main/kotlin/io/provenance/engine/service/JobService.kt:19]: polling for jobs and then dispatching them to the appropriate handlers (kind of like event retry, but a bit different)

File: p8e-api/src/main/kotlin/io/provenance/engine/crypto/Account.kt [hunk 3]
```diff
@@ -2,6 +2,8 @@ package io.provenance.engine.crypto
 
 import io.p8e.crypto.Hash
 import io.p8e.crypto.Numeric
+import io.provenance.engine.crypto.Bech32
+import io.provenance.engine.crypto.toBech32Data
```
Comment (rchaing-figure) [inline p8e-api/src/main/kotlin/io/provenance/engine/crypto/Account.kt:6]: these can probably go away

**Review by rchaing-figure — APPROVED** [review]
few questions and comments

File: os-proto/src/main/proto/objectstore/public_key.proto [hunk 4]
```diff
@@ -14,14 +14,22 @@ service PublicKeyService {
...
     string url = 2;
+    // optional - contains any associated metadata that the caller wants to attach
+    // to this public key
     google.protobuf.Any metadata = 3;
 }
 
 message PublicKeyResponse {
     UUID uuid = 1;
     PublicKey public_key = 2;
+    PublicKey signing_public_key = 7;
```
[the remark this answers is not in the document]
Comment (piercetrey-figure) [reply os-proto/src/main/proto/objectstore/public_key.proto:32]: I think os is now just verifying the signature of object put requests

File: os-proto/src/main/proto/objectstore/public_key.proto [hunk 5]
```diff
@@ -14,14 +14,22 @@ service PublicKeyService {
...
+    // when blank, this public key is associated with the local p8e environment
     string url = 2;
+    // optional - contains any associated metadata that the caller wants to attach
+    // to this public key
     google.protobuf.Any metadata = 3;
 }
 
 message PublicKeyResponse {
     UUID uuid = 1;
     PublicKey public_key = 2;
```
[the remark this answers is not in the document]
Comment (piercetrey-figure) [reply os-proto/src/main/proto/objectstore/public_key.proto:31]: yes, these protos are just copied over from object store, so I didn't change the name to clarify

**Comment by piercetrey-figure** [comment]
:rip:

**Pull request closed without merging**